### ISSUE TITLE
Training Multi-band MelGAN gets huge loss value and cannot recover

### ISSUE THREAD

**Issue by ttsking** [TensorSpeech/TensorFlowTTS#646]
I use a customized Baker Dataset to train the Multi-band MelGAN model. It's fine at the beginning and the lost value suddenly goes huge.

I tested training from scratch:
![tensorboard_from scratch](https://user-images.githubusercontent.com/74232385/129327438-0c05569a-f843-4170-b66d-3b0fb63b015e.jpeg)

and Fine-tune:
![tensorboard_finetune](https://user-images.githubusercontent.com/74232385/129327437-0439797e-d895-46e9-b534-b58099d35288.jpeg)

If i set --generator_mixed_precision 1, the loss value become NaN. If i set  --generator_mixed_precision 0, i can get the loss value, but it is larger than 3000:

```
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_adversarial_loss = 0.0000.
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_spectral_convergence_loss = 7186.2988.
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_log_magnitude_loss = 2.5740.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_spectral_convergence_loss = 52.8283.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_log_magnitude_loss = 2.9583.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_gen_loss = 3622.3291.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_real_loss = 0.0000.
2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_fake_loss = 0.0000.
2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_dis_loss = 0.0000.
```

**Comment by ttsking** [comment]
Update: When this error occured, i use a customized Baker Dataset in sampling 24000. After that, i resampled the dataset to 48000 like the standard Baker Dataset. Then the training gets successful.  I don't know why.

![compare-24-48](https://user-images.githubusercontent.com/74232385/129378074-03a6cada-b8eb-4b59-b8ea-6a50a3b6670e.png)
